platform/qemu.go

```diff
@@ -199,10 +216,10 @@ func setupDisk(imageFile string) (*os.File, error) {
 	cp.Stderr = os.Stderr
 
 	if err := cp.Run(); err != nil {
-		return nil, err
+		return "Error: fail to copy the image file", err
 	}
 
-	return os.OpenFile(dstFileName, os.O_RDWR, 0)
+	return dstFileName, err
```

Comment:
err isn't meaningful at this point and is going to be a left over nil from ioutil.TempFile. Just do `return dstFileName, nil`